### ISSUE TITLE
Library Function - Edit Audio Markers does not work

### ISSUE THREAD

**Issue by BeDazzleMe** [olsson82/rivendellweb#23]
Running on ubuntu with RD 4.3.
In Library -> Edit Audio Markers does not work.

It tries to load the cart but never does.
![rdweb-library-edit-audio-markers](https://github.com/user-attachments/assets/cd0d79c6-8b83-47ca-b079-bf4febc5c1dc)

**Comment by olsson82** [comment]
That should work.
I can be some server settings that you need to adjust in php.ini file.

Try to change your php.ini settings with the following settings:

max_execution_time 180
memory_limit 128M
post_max_size 200M
upload_max_filesize 200M
max_input_time 60
max_input_vars 3000

**Comment by BeDazzleMe** [comment]
Hi Andreas,

I updated /etc/php/8.1/apache2/php.ini with the settings provided and restarted apache2, however it still does not work.

I also rebooted the ubuntu VM and started the web gui again, but no change.

Darren.

**Comment by olsson82** [comment]
That part uses Rivendell API to get the audio. It's works fine for me on my test machine, and also on my live production machine.

Do you have set access to all audio function on the user in RDAdmin ?

**Comment by olsson82** [comment]
You can also check in /var/log/apache2/error.log if it report any errors.

**Comment by BeDazzleMe** [comment]
Yes, this appears in the apache logs when I attempt to edit audio markers:

[Thu Aug 29 08:55:04.716080 2024] [php:error] [pid 702] [client 172.25.3.99:51943] PHP Fatal error:  Uncaught TypeError: Unsupported operand types: string * int in /var/www/html/forms/library/export.php:44\nStack trace:\n#0 /var/www/html/forms/library/export.php(53): getDurationOfWavInMs()\n#1 {main}\n  thrown in /var/www/html/forms/library/export.php on line 44, referer: http://172.25.5.170/library/carts/cart/audio/28041
[Thu Aug 29 08:55:04.781449 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:05.782797 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:06.784184 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:07.785576 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:08.786922 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:09.788361 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:10.789813 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:11.791340 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:12.792934 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:13.794477 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:14.795993 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:15.797389 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:16.798705 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80
[Thu Aug 29 08:55:17.800015 2024] [core:warn] [pid 649] (99)Cannot assign requested address: AH00056: connect to listener on [::]:80

**Comment by BeDazzleMe** [comment]
> That part uses Rivendell API to get the audio. It's works fine for me on my test machine, and also on my live production machine.
> 
> Do you have set access to all audio function on the user in RDAdmin ?

Yes.

**Comment by olsson82** [comment]
Thanks
Did you install ffmpeg package on the server ?

`sudo apt install ffmpeg -y`

For the moment that is needed to calculate the length of the audio file.

**Comment by BeDazzleMe** [comment]
yes, that was part of the installation instructions and was completed.

**Comment by BeDazzleMe** [comment]
I assume I don't need to configure jack or any audio on the VM running ubuntu + RD 4.3.

**Comment by olsson82** [comment]
Ok i can see it complains on string.

Do you use the regular cart numbers that are in Rivendell or do you use some custom cart names ?

**Comment by BeDazzleMe** [comment]
I use the standard cart numbers in RD.

**Comment by olsson82** [comment]
I can see that it has some problems to read the length of the file.

Do you store the audio files in /var/snd folder ?

**Comment by BeDazzleMe** [comment]
Yes... and I've just found the problem - NFS permissions from the ubuntu test node were not open on the RD4 filer machine.

Apologies, that was my bad.

Now working!
![rdweb-edit-audio-working](https://github.com/user-attachments/assets/b6acd3a2-5032-4560-a85e-c6c67b7a6c3e)

**Comment by olsson82** [comment]
No problem. Good to see it working.

**Comment by BeDazzleMe** [comment]
Looks good, you've done a great job on this. I am happy to test and provide feedback.

**Comment by olsson82** [comment]
Thanks, give me feedback, reoort bugs, and wishes. There are alot that i can have forget about.
